### PR TITLE
Document test_driver.set_permission().

### PR DESCRIPTION
**Pull request by rakuco**

### CODE REVIEW

File: docs/writing-tests/testdriver.md [hunk 1]
```diff
@@ -128,3 +128,27 @@ To send special keys, one must send the respective key's codepoint. Since this u
 For example, to send the tab key you would send "\uE004".
 
 [activation]: https://html.spec.whatwg.org/multipage/interaction.html#activation
+
+### set_permission
+
+Usage: `test-driver.set_permission(descriptor, state, one_realm)`
+ * `descriptor`: a
```
Comment (sideshowbarker) [inline docs/writing-tests/testdriver.md]:
```suggestion
 * _descriptor_: a
```

File: docs/writing-tests/testdriver.md [hunk 2]
```diff
@@ -128,3 +128,27 @@ To send special keys, one must send the respective key's codepoint. Since this u
...
+   [PermissionDescriptor](https://w3c.github.io/permissions/#dictdef-permissiondescriptor)
+   or derived object
+ * `state`: a
```
Comment (sideshowbarker) [inline docs/writing-tests/testdriver.md]:
```suggestion
 * _state_: a
```

File: docs/writing-tests/testdriver.md [hunk 3]
```diff
@@ -128,3 +128,27 @@ To send special keys, one must send the respective key's codepoint. Since this u
...
+   [PermissionState](https://w3c.github.io/permissions/#enumdef-permissionstate)
+   value
+ * `one_realm`: a boolean that indicates whether the permission settings
```
Comment (sideshowbarker) [inline docs/writing-tests/testdriver.md]:
```suggestion
 * _one_realm_: a boolean that indicates whether the permission settings
```

File: docs/writing-tests/testdriver.md [hunk 4]
```diff
@@ -128,3 +128,27 @@ To send special keys, one must send the respective key's codepoint. Since this u
...
+   apply to only one realm
+
+This function causes permission requests and queries for the status of a
+certain permission type (e.g. "push", or "background-fetch") to always
+return `state`. It returns a `Promise` that resolves after the
```
Comment (sideshowbarker) [inline docs/writing-tests/testdriver.md]:
```suggestion
return _state_. It returns a Promise that resolves after the
```

File: docs/writing-tests/testdriver.md [hunk 5]
```diff
@@ -128,3 +128,27 @@ To send special keys, one must send the respective key's codepoint. Since this u
...
+permission has been to be overriden with `state`.
```
Comment (sideshowbarker) [inline docs/writing-tests/testdriver.md]:
```suggestion
permission has been to be overridden with _state_.
```

**Review by sideshowbarker — COMMENTED** [review]
"descriptor", "state", and "one_realm" should marked up so as to be rendered in italics rather than monospace (since they are not literal names but instead are references to parameters/values)

File: docs/writing-tests/testdriver.md (hunk 1 above)
Comment (rakuco) [reply docs/writing-tests/testdriver.md]: Fixed.

File: docs/writing-tests/testdriver.md (hunk 2 above)
Comment (rakuco) [reply docs/writing-tests/testdriver.md]: Fixed.

File: docs/writing-tests/testdriver.md (hunk 3 above)
Comment (rakuco) [reply docs/writing-tests/testdriver.md]: Fixed.

File: docs/writing-tests/testdriver.md (hunk 4 above)
Comment (rakuco) [reply docs/writing-tests/testdriver.md]: Fixed.

File: docs/writing-tests/testdriver.md (hunk 5 above)
Comment (rakuco) [reply docs/writing-tests/testdriver.md]: Fixed.

**Comment by rakuco** [comment]
> "descriptor", "state", and "one_realm" should marked up so as to be rendered in italics rather than monospace (since they are not literal names but instead are references to parameters/values)

Thanks for the review. Do note though that the rest of this file probably needs to be updated too -- I was following the existing style when writing this bit.

**Comment by sideshowbarker** [comment]
> > "descriptor", "state", and "one_realm" should marked up so as to be rendered in italics rather than monospace (since they are not literal names but instead are references to parameters/values)
> 
> Thanks for the review. Do note though that the rest of this file probably needs to be updated too -- I was following the existing style when writing this bit.

aha — OK, I can raise a separate PR with a patch for the styling of the existing parts of the file that are marked up wrong. (And we don’t need to let that block this PR from getting merged in the meantime.)

**Review by sideshowbarker — APPROVED** [review]

File: docs/writing-tests/testdriver.md [hunk 6]
```diff
@@ -128,3 +128,27 @@ To send special keys, one must send the respective key's codepoint. Since this u
...
+ * _state_: a
+   [PermissionState](https://w3c.github.io/permissions/#enumdef-permissionstate)
+   value
+ * _one_realm_: a boolean that indicates whether the permission settings
+   apply to only one realm
+
+This function causes permission requests and queries for the status of a
+certain permission type (e.g. "push", or "background-fetch") to always
+return _state_. It returns a Promise that resolves after the permission has
+been to be overridden with _state_.
```
Comment (LukeZielinski) [inline docs/writing-tests/testdriver.md]: seems like a typo here. Perhaps you meant "...after the permission has been **set** to be overridden..."?

**Review by LukeZielinski — APPROVED** [review]
Thanks! Just one nit.

File: docs/writing-tests/testdriver.md (hunk 6 above)
Comment (rakuco) [reply docs/writing-tests/testdriver.md]: That does make much more sense :-) Thanks, I've pushed a fixed version.

**Pull request merged**